src/constants.ts

```diff
@@ -6,6 +6,14 @@ export const MAX_BIP_44_DEPTH = 5;
 export const MAX_UNHARDENED_BIP_32_INDEX = 0x7fffffff; // 2^31 - 1
 export const MAX_BIP_32_INDEX = 0xffffffff; // 2^32 - 1
 
+/**
+ * A guard symbol to prevent untrusted public keys from being passed to
+ * `SLIP10Node` constructors.
+ */
+export const PUBLIC_KEY_GUARD = Symbol(
```

Comment:
Right now this is exported in the index 😄

Reply:
https://github.com/MetaMask/key-tree/blob/cc54abfd15b47abaa9feafaf3bd6488da306bebe/src/index.ts#L31

Reply:
I wrongfully assumed we only did explicit exports instead of wildcard exports. Will move the guard to a separate file.

Reply:
Probably worth testing in the index.test that it isn't exported

Reply:
Moved to `guard.ts` and added a test to ensure nothing from `guard.ts` is exported.